core/stores/mon/bulkinserter.go

```diff
@@ -42,6 +42,26 @@ func NewBulkInserter(coll *mongo.Collection, interval ...time.Duration) *BulkIns
 	}
 }
 
+// NewBatchInserter returns a BulkInserter.
+func NewBatchInserter(coll Collection, interval ...time.Duration) *BulkInserter {
```

Comment:
why not just call NewBulkInserter here? And put the implementation of NewBulkInserter into a new func.